EmbeddedDocumentsQuery/testsrc/EmbeddedDocumentsQueryTest.java

```diff
@@ -0,0 +1,84 @@
+import static org.junit.Assert.*;
+
+import java.util.List;
+import org.bson.Document;
+import org.junit.Test;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+
+
+public class EmbeddedDocumentsQueryTest {
+    
+	static String CONNECTION_STRING = "mongodb+srv://Test:Test987#@cluster0-jxlpi.mongodb.net/test";
+	
+	@Test
+	public void TestConnection() {		
+		MongoClientURI clientUri = new MongoClientURI(CONNECTION_STRING);
+		try(MongoClient client = new MongoClient(clientUri))
+		{
+			assertNotNull(client);
+		}    
+	}
+
+	@Test
+	public void SingleNestedFieldEqualityMatchTest() {
+		EmbeddedDocumentsQuery embeddedDocumentsQuery = new EmbeddedDocumentsQuery();		
+	    List<Document> filteredDocuments = embeddedDocumentsQuery.SingleNestedFieldEqualityMatch(CONNECTION_STRING,"Woodburn");
+	    
+	    //verify size
+	    assertEquals(1, filteredDocuments.size());	
+	    
+	    //verify record
+	    Document result = filteredDocuments.get(0);
+	    String id = result.getString("_id");
+	    assertEquals("5a989e4abf545f22ed33d590", id);	    
+	}
+	
+	
+	@Test
+	public void TwoNestedFieldsEqualityMatchTest() {
```

Comment:
This test needs to test more possible inputs such as an empty string, a value that results in no matches, a value that results in exactly one match, and a value that results in multiple documents matched. You need to test all cases so that we know your example works in the face of changes to the underlying dataset.